src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -783,6 +791,7 @@ def write_dataset(self, **kwargs):  # noqa: C901
         options['dtype'] = builder.dtype
         dset = None
         link = None
+        print('    writing hdf5 dataset', builder.name, link_data, type(data))
```

Comment:
All these debug prints should be removed. We should consider using Python's logging module for the I/O-related parts of HDMF. I think that could help with debugging.